### PR TITLE
feature(API): Adds input validation in all API methods and validates empty user Id.

### PR DESCRIPTION
**Pull request by mfahadahmed**

### CODE REVIEW

**Comment by msohailhussain** [comment]
build

File: OptimizelySDK/Optimizely.cs [hunk 1]
```diff
@@ -337,7 +339,13 @@ public Variation GetVariation(string experimentKey, string userId, UserAttribute
...
         public bool SetForcedVariation(string experimentKey, string userId, string variationKey)
         {
-            return Config.SetForcedVariation(experimentKey, userId, variationKey);
+            var inputValues = new Dictionary<string, string>
+            {
+                { USER_ID, userId },
+                { EXPERIMENT_KEY, experimentKey }
+            };
+
+            return ValidateStringInputs(inputValues) && Config.SetForcedVariation(experimentKey, userId, variationKey);
```
Comment (msohailhussain) [inline OptimizelySDK/Optimizely.cs]: it's not readable, make sure it's readable.

File: OptimizelySDK/Optimizely.cs [hunk 2]
```diff
@@ -337,7 +339,13 @@ public Variation GetVariation(string experimentKey, string userId, UserAttribute
         /// <returns>A boolean value that indicates if the set completed successfully.</returns>
         public bool SetForcedVariation(string experimentKey, string userId, string variationKey)
         {
-            return Config.SetForcedVariation(experimentKey, userId, variationKey);
+            var inputValues = new Dictionary<string, string>
```
Comment (msohailhussain) [inline OptimizelySDK/Optimizely.cs]: If we are using it for input values, please name variable name correctly, the actual purpose of the variable.

**Comment by mikeng13** [comment]
build

**Pull request merged**